### ISSUE TITLE
Impossible to get recommonmark to work with sphinx

### ISSUE THREAD

**Issue by guyzmo** [rtfd/recommonmark#50]
it might not directly be a bug, but an issue using recommonmark: I just cannot get it to work! It parses all my markdown files like it is ReST text… And I cannot find why it's not working.

Though as I've looked up through every bit of RTFM available, I thought it could be a good idea to post here so at least there's a place where it gets documented.

When an answer is so hard to find for an issue, I know it usually means it's a PEBCAK issue, and there's something very stupid I forgot to do or setup, and I'll feel stupid when we'll find that. But still, I cannot find what that is so… 

A summary of what I've done:

* I am downloading Sphinx+recommonmark at latest versions in my [requirements.txt].
* I configured sphinx so [conf.py] features [recommonmark setup]
* I wrote my markdown files in [doc] directory and my docstrings generated as rst.
* building it throws a lot of [errors and warnings] — I get the same locally or on rtfd.
* which end up being that [mess] when I expect something like [that].
* and the rst files [show up fine].

So please, please, can anybody tell me what I'm doing wrong here?

N.B.: I use python3 only (did not try python2).

[requirements.txt]:https://github.com/guyzmo/buildstrap/blob/master/requirements-doc.txt
[conf.py]:https://github.com/guyzmo/buildstrap/blob/master/doc/conf.py
[recommonmark setup]:https://github.com/guyzmo/buildstrap/blob/master/doc/conf.py#L25-42
[doc]:https://github.com/guyzmo/buildstrap/blob/master/doc/
[errors and warnings]:https://readthedocs.org/projects/buildstrap/builds/4101578/
[mess]:https://buildstrap.readthedocs.io/en/latest/
[that]:https://github.com/guyzmo/buildstrap/blob/master/doc/index.md
[show up fine]:http://buildstrap.readthedocs.io/en/latest/buildstrap.html

**Comment by pfultz2** [comment]
It seems like recommonmark doesn't support [link references](http://spec.commonmark.org/0.25/#link-reference-definitions) that well.

**Comment by guyzmo** [comment]
ok, but if it's only an issue with link references, wouldn't I get an error from recommonmark? or at _least_ those links poorly formatted (and then I'd just need to fix that). Right now it's just parsing markdown as ReST, which is pretty obvious given that the triple back quote is interpreted as a `<pre>` block with one back quote within (as the double backquote is the blockquote in ReST). And I'm getting warning at about every token.

So I really think that recommonmark does not get called!

I actually even added printouts in about every function in recommonmark, and it does not appear when sphinx calls it. I also tried playing around with the `sys.path` with no effect. Though, Sphinx has access to `recommonmark`, because in `conf.py` I'm not getting an `ImportError`!

**Comment by pfultz2** [comment]
> I actually even added printouts in about every function in recommonmark, and it does not appear when sphinx calls it.

Sphinx may be finding a different version of recommonmark than the one you have annotated with print statements or sphinx is suppressing the output, but sphinx definitely calls recommonmark. I have written integration tests [here](https://github.com/pfultz2/recommonmark/blob/develop/tests/test_sphinx.py) that shows it does get called.

**Comment by guyzmo** [comment]
ok...

I have just lost five hours of my life for a stupid mistake. As I started my issue I said *PEBCAK* and *stupid* issue.

Well... that's what it was. I wrote 

    parsers = {...}

instead of

    source_parsers = {...}

no wonder why it wasn't working. And I've reread that file 50 times, and did not catch that a single time. I eventually found out as I copied/pasted one of the test examples into my code directly, and then had it working, and then looked for differences.

Maybe adding to sphinx a summary of the `conf.py` options that are taken into account would be a nice idea.

Thank you for your help!
